src/mono/mono/metadata/sgen-mono.c

```diff
@@ -2639,7 +2639,7 @@ mono_gc_get_allocated_bytes_for_current_thread (void)
 	info = mono_thread_info_current ();
 
 	/*There are some more allocated bytes in the current tlab that have not been recorded yet */
-	return info->total_bytes_allocated + info->tlab_next - info->tlab_start;
+	return info->total_bytes_allocated + (ptrdiff_t)(info->tlab_next - info->tlab_start);
```

Comment:
Actually this change does nothing anyway.

integer + pointer - pointer
is (integer + pointer) - pointer
is pointer - pointer
is ptrdiff_t.

Reply:
Oh, because total_bytes_allocated is 64bits but gets truncated to 32?
Ok, but the cast is still not needed.